lang/java/reef-common/src/main/java/org/apache/reef/driver/parameters/NodeLabelExpression.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.driver.parameters;
+
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+
+/**
+ * Node label expression string.
+ */
+@NamedParameter(doc="Node label expression")
+public final class NodeLabelExpression implements Name<String> {
```

Comment:
I used `NodeLabelExpression` because it is the term YARN uses. But I think `NodeLabel` is more appropriate if people think so. Given that this parameter is for not only YARN but other RMs such as Mesos and Kubernetes, we don't have to follow YARN's term.